lib/grape/request.rb

```diff
@@ -2,27 +2,39 @@ module Grape
   class Request < Rack::Request
     HTTP_PREFIX = 'HTTP_'.freeze
 
+    alias_method :rack_params, :params
+
     def params
-      @params ||= begin
-        params = Hashie::Mash.new(super)
-        if env[Grape::Env::RACK_ROUTING_ARGS]
-          args = env[Grape::Env::RACK_ROUTING_ARGS].dup
-          # preserve version from query string parameters
-          args.delete(:version)
-          args.delete(:route_info)
-          params.deep_merge!(args)
-        end
-        params
-      end
+      @params ||= _build_params
     end
 
     def headers
-      @headers ||= env.each_with_object({}) do |(k, v), h|
+      @headers ||= _build_headers
+    end
+
+    private
+
+    def _build_params
```

Comment:
Why these underscore methods? I think to be consistent they should just be `build_params` and `build_headers`? Not to introduce another convention ...